models/Day.js

```diff
@@ -1,16 +1,38 @@
 import mongoose from "mongoose";
+import nanoid from "nanoid";
 
+// let id = mongoose.Types.ObjectId();
 const { Schema } = mongoose;
 // const weightSchema = new Schema({ type: Number });
 // const heightSchema = new Schema({ type: Number });
 // const feastTimeSchema = new Schema({ type: String });
 
+// [{ timeStamp: { type: String }, value: { type: Number }, id: nanoid() }];
+
 const daySchema = new Schema(
   {
     date: { type: String, required: true },
-    weight: [Number],
-    height: [Number],
-    feastTime: [String],
+    weights: [
+      {
+        timeStamp: { type: String },
+        value: { type: Number },
+        id: { type: mongoose.Types.ObjectId },
+      },
+    ],
+    heights: [
+      {
+        timeStamp: { type: String },
+        value: { type: Number },
+        id: { type: mongoose.Types.ObjectId },
+      },
+    ],
+    feastTimes: [
+      {
+        timeStamp: { type: String },
+        value: { type: String },
+        id: { type: mongoose.Types.ObjectId },
+      },
+    ],
```

Comment:
nice! if you specify this type ObjectId, does mongoose generate the id automatically for you?

Reply:
yes it does, but you need to call with ```object._id``` , apparently it doesn't work with just ```object.id```